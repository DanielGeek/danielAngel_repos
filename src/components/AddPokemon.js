import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddPokemon = ({ setPokemones }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value.toLowerCase());
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      setPokemones( pokemones => {
        const newpokemones = pokemones.filter( pokemon => pokemon !== inputValue)
        return [ inputValue, ...newpokemones]
      });
      setInputValue('');
    }

  }

  return (
    <>
      <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
        <input
          type="text"
          placeholder='Buscar'
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </>
  )
}

AddPokemon.propTypes = {
  setPokemones: PropTypes.func.isRequired
}
