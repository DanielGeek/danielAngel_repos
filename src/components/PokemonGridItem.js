import React from 'react';
import PropTypes from 'prop-types';
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLette';

export const PokemonGridItem = ({ name, url, id }) => {

  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={ url } alt={ name } />
      <p># { id }</p>
      <p>{ capitalizeFirstLetter(name) }</p>
    </div>
  )
}

PokemonGridItem.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}
