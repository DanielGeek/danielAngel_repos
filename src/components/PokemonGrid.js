import React from 'react'
import PropTypes from 'prop-types';

import { useFetchPokemon } from '../hooks/useFetchPokemon'
import { PokemonGridItem } from './PokemonGridItem';

export const PokemonGrid = ({ pokemonNameOrId }) => {

  const { data: images, loading, error } = useFetchPokemon( pokemonNameOrId );

  let imagesFilters = images.filter(image => image.id !== pokemonNameOrId);

  return (
    <>

      { loading && <p className='animate__animated animate__flash'>Cargando</p>}
      { !error &&
          <div className='card-grid'>
          {
            imagesFilters.map( img => (
              <PokemonGridItem
                  key={ img.id }
                  { ...img }
              />
            ))
          }
        </div>
      }
    </>
  )
}

PokemonGrid.propTypes = {
  pokemonNameOrId: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}