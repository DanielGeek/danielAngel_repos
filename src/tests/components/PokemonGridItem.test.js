import React from 'react';
import { shallow } from "enzyme";
import { PokemonGridItem } from "../../components/PokemonGridItem";

describe('Pruebas en <PokemonGridItem />', () => {

  const name = 'un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<PokemonGridItem name={name} url={url} />)

  test('debe de mostrar el componente correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(name);
  });

  test('debe de tener animate_fadeIn', () => {

    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });
})