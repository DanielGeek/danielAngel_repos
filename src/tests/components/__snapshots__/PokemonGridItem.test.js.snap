// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PokemonGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    # 
  </p>
  <p>
    Un titulo
  </p>
</div>
`;
